src/components/frame/Extensions/layouts/hooks.tsx

```diff
@@ -357,48 +357,52 @@ export const useGetTxButtons = () => {
   const { groupState } = useUserTransactionContext();
   const isMobile = useMobile();
 
-  const txButtons = [
-    groupState === TransactionGroupStates.SomePending && (
-      <TxButton
-        text={isMobile ? undefined : { id: 'button.pending' }}
-        className={clsx({
-          '!min-w-0': isMobile,
-        })}
-        icon={
-          <span
-            className={clsx('flex shrink-0', {
-              'ml-1.5': !isMobile,
-            })}
-          >
-            <Icon
-              name="spinner-gap"
-              className="animate-spin"
-              appearance={{ size: 'tiny' }}
-            />
-          </span>
-        }
-        data-openhubifclicked // see UserReputation for usage
-      />
-    ),
-    groupState === TransactionGroupStates.AllCompleted && (
-      <TxButton
-        text={isMobile ? undefined : { id: 'button.completed' }}
-        className={clsx({
-          '!min-w-0': isMobile,
-        })}
-        icon={
-          <span
-            className={clsx('flex shrink-0', {
-              'ml-1.5': !isMobile,
-            })}
-          >
-            <Icon name="white-tick" appearance={{ size: 'tiny' }} />
-          </span>
-        }
-        data-openhubifclicked // see UserReputation for usage
-      />
-    ),
-  ];
+  const txButtons = (
```

Comment:
I'd much rather see this be a component instead of a hook returning JSX so we avoid antipatterns

Reply:
@bassgeta 
I suggest to refactor it in separate pull request, not here. This pull request is responsible to display error messages and that's all.

Reply:
makes sense yeah